src/api/v2/ResourceGroupsExtensionManager.ts

```diff
@@ -27,4 +40,37 @@ export class ResourceGroupsExtensionManager {
             }
         }
     }
+
+    async activateApplicationResourceProviders(): Promise<void> {
+        const inActiveResourceContributors =
+            vscode.extensions.all
+                .map(extension => ({ extension, resources: (extension.packageJSON as ExtensionPackage)?.contributes?.['x-azResourcesV2']?.application?.resources ?? [] }))
+                .filter(extensionAndContributions => extensionAndContributions.resources.length > 0 && !extensionAndContributions.extension.isActive);
+
+        await Promise.all(inActiveResourceContributors.map(contributor => contributor.extension.activate()));
+    }
+
+    async activateWorkspaceResourceProviders(): Promise<void> {
```

Comment:
Workspace resource providers might not really need an activation mechanism beyond the standard VSCode `onView:...`. Since they will always be providing nodes at the root of the tree, they always need to be activated immediately (in contrast to the remote resource view, which would start with their node collapsed a few layers down)

Reply:
That's true, though that would mean coupling those extensions with our arbitrary ID for our resources/workspaces view, *in addition* to registering themselves as resource providers and/or BDPs.  It seems better to have extensions worry only about their specific registrations.

Unless you're suggesting that workspaces providers not have *any* static registration at all, relying on VS Code view activation and the resulting API provider registration call?  While that *could* work, it still requires registration (again, exposing our specific view ID) that would be very different than BDP registration.  I feel like we'd want registration to be consistent between resources/workspaces and provider/BDP.

Reply:
That's fair. Consistency is probably a worthwhile benefit to streamlining activation through this mechanism.